### DISCUSSION TITLE
Why does "Separator" need "use client"?

### DISCUSSION THREAD

**Discussion by avloss** [shadcn-ui/ui#2562] [Q&A]
I'm trying to wrap my head around how to develop for Next.js 13+, and it's my understanding that "use client" should be avoided when possible, only reserving it for user interactions. I understand that still it might be used for animations, etc. But "Separator" element surely doesn't fit in a category where one would expect that JS would be needed to render it.

I'm still learning about this new paradigm, and perhaps an answer would be helpful to other too. Initial thought would be that it's just "bloated", but then there's probably a good reason for why it needs to run JS on a client to render a separator.

**Comment by avloss** [comment]
https://github.com/shadcn-ui/ui/blob/a465432a664ab41b36bca970fdcad2341c873f6d/apps/www/registry/default/ui/separator.tsx#L1

I've tried referencing this line while creating this question, but then it seemed not to have worked.

**Comment by Alexjoshua14** [comment]
Hey @avloss, the reason that `'use client'` is needed for the Separator component is because it depends on Radix-ui's Separator component which relies on their Primitive component which uses `use effect`.  

Essentially when the Separator component (or more specifically the Primitive component) is initially mounted, `window['radix-ui']` is set to true. 

In general, it's my understanding that components should be client components whenever user interactivity is needed or information about the client environment (such as using `window`) is needed.

Here's some snippets of the code pieces with permalinks embedded that are weaved together. 

[<img width="801" alt="Screenshot 2024-01-25 at 11 17 41 AM" src="https://github.com/shadcn-ui/ui/assets/59298565/a35ddeb2-983a-4dc0-afdc-8b8e38610429">](https://github.com/radix-ui/primitives/blob/c31c97274ff357aea99afe6c01c1c8c58b6356e0/packages/react/separator/src/Separator.tsx#L39)


[<img width="853" alt="Screenshot 2024-01-25 at 11 12 50 AM" src="https://github.com/shadcn-ui/ui/assets/59298565/49a0efdb-3b47-4cf4-ac15-841ce7c56cf2">](https://github.com/radix-ui/primitives/blob/c31c97274ff357aea99afe6c01c1c8c58b6356e0/packages/react/primitive/src/Primitive.tsx#L49)
